File: app/src/main/java/com/me/bui/architecturecomponents/di/ActivityBuildersModule.java
package com.me.bui.architecturecomponents.di;

import com.me.bui.architecturecomponents.MainActivity;
import com.me.bui.architecturecomponents.ui.RepoFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by mao.bui on 9/3/2018.
 */

@Module
public abstract class ActivityBuildersModule {

    @ContributesAndroidInjector(modules = FragmentBuildersModule.class)
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/me/bui/architecturecomponents/di/ActivityBuildersModule.kt
package com.me.bui.architecturecomponents.di

import com.me.bui.architecturecomponents.MainActivity
import com.me.bui.architecturecomponents.ui.RepoFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by mao.bui on 9/3/2018.
 */

@Module
abstract class ActivityBuildersModule {

    @ContributesAndroidInjector(modules = arrayOf(FragmentBuildersModule::class))
    abstract fun contributeMainActivity(): MainActivity
}
